src/test/java/com/snowplowanalytics/redash/RedashClientDataSourceTest.java

```diff
@@ -0,0 +1,243 @@
+
+    @Test(expected = IOException.class)
+    public void addDataSourceToGroupWithWrongClientTest() throws IOException {
+        int id = redashClient.createDataSource(rds);
+        wrongClient.addDataSourceToGroup(id, defaultGroup.getId());
+    }
+
+    @Test
+    public void removeDataSourceFromGroupTest() throws IOException {
+        int id = redashClient.createDataSource(rds);
+        DataSource fromDb = redashClient.getDataSource(rds.getName());
+        Assert.assertTrue(fromDb.getGroups().size() == 1);
+        Assert.assertTrue(redashClient.removeDataSourceFromGroup(id, defaultGroup.getId()));
+        Assert.assertTrue(redashClient.getDataSource(rds.getName()).getGroups().isEmpty());
+        Assert.assertFalse(redashClient.removeDataSourceFromGroup(id, defaultGroup.getId()));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void removeNonExistingDataSourceFromGroupTest() throws IOException {
+        int id = redashClient.createDataSource(rds);
+        DataSource fromDb = redashClient.getDataSource("wrongName" + rds.getName());
+        Assert.assertTrue(fromDb.getGroups().size() == 1);
+        Assert.assertTrue(redashClient.removeDataSourceFromGroup(id + 1, defaultGroup.getId()));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void removeDataSourceFromNonExistingGroupTest() throws IOException {
+        int id = redashClient.createDataSource(rds);
+        DataSource fromDb = redashClient.getDataSource(rds.getName());
+        Assert.assertTrue(fromDb.getGroups().size() == 1);
+        Assert.assertTrue(redashClient.removeDataSourceFromGroup(id, defaultGroup.getId() + 1));
+    }
+
+    @Test(expected = IOException.class)
+    public void removeDataSourceFromGroupWithWrongClientTest() throws IOException {
+        int id = redashClient.createDataSource(rds);
+        wrongClient.removeDataSourceFromGroup(id, defaultGroup.getId());
+    }
+
+    public static boolean dataSourceMatcher(DataSource first, DataSource second) {
+        if (first.getName() == null || first.getType() == null) return false;
+        return first.getName().equals(second.getName()) &&
+                first.getType().equals(second.getType()) &&
+                first.getHost().equals(second.getHost()) &&
+                first.getPort() == second.getPort() &&
+                first.getDbName().equals(second.getDbName()) &&
+                first.getUser().equals(second.getUser());
+    }
+
+    private boolean simpleDatasourceMatcher(DataSource dataSource) {
+        if (dataSource.getName() == null || dataSource.getType() == null) return false;
+        return dataSource.getName().equals(rds.getName()) && dataSource.getType().equals(rds.getType());
+    }
+
+    private void wipeDataSources() throws IOException {
+        redashClient.getDataSources().forEach(dataSource -> {
+            try {
+                redashClient.deleteDataSource(dataSource.getId());
+            } catch (IOException e) {
```

Comment:
Is this doing anything after catching the exception? It's confusing because the method indicates is can also throw an IOException